verilog/formatting/formatter_test.cc

```diff
@@ -1382,6 +1438,53 @@ static const std::initializer_list<FormatterTestCase> kFormatterTestCases = {
      "  f = x;\n"
      "  // statement comment\n"  // indented
      "endfunction\n"},
+    // function arguments indenting/wrapping
+    {"class c; function check(core_status_t core_status,"
+     " string error_msg, int timeout); endfunction; endclass;",
+     "class c;\n"
+     "  function check(\n"
+     "    core_status_t core_status,\n"
+     "    string error_msg,\n"
+     "    int timeout);\n"
+     "  endfunction;\n"
+     "endclass;\n"},
+    {"class c; function check(core_status_t core_status,"
+     " string error_msg); endfunction; endclass;",
+     "class c;\n"
+     "  function check(\n"
+     "    core_status_t core_status,\n"
+     "    string error_msg);\n"
+     "  endfunction;\n"
+     "endclass;\n"},
+    {"class c; function check(core_status_t core_status"
+     " ); endfunction; endclass;",
+     "class c;\n"
+     "  function check(\n"
+     "    core_status_t core_status);\n"
+     "  endfunction;\n"
+     "endclass;\n"},
+    // test correctness of fitting 1st function argument
+    {"class c; function check(core_t core,"
+     " string error_msg, int timeout); endfunction; endclass;",
+     "class c;\n"
+     "  function check(core_t core,\n"
+     "                 string error_msg,\n"
+     "                 int timeout);\n"
+     "  endfunction;\n"
+     "endclass;\n"},
+    {"class c; function check(core_t core,"
+     " string error_msg); endfunction; endclass;",
+     "class c;\n"
+     "  function check(core_t core,\n"
+     "                 string error_msg);\n"
+     "  endfunction;\n"
+     "endclass;\n"},
+    {"class c; function check(core_t core"
+     " ); endfunction; endclass;",
+     "class c;\n"
+     "  function check(core_t core);\n"
+     "  endfunction;\n"
+     "endclass;\n"},
```

Comment:
Do you have an example that formats to:

```
function f(loooooooooooooooooong_type a,
           short_type b, short_type c);
endfunction
```

Specifically, compacting multiple sub-partitions on the same line when they fit.
If not yet implemented, is it easy to add?  (can also punt to follow-up PR too)

Reply:
I didn't want to fit sub-partitions if they fit. I wanted to fit all in one line or expand it all.
It's not hard to add. It could be done as you told in earlier comments:
1. Add new police, e.g. `kAppendFittingSubpartitions`
2. During partition expansion phase treat it as `kExpandAll`
3. Later in formatting phase apply greedy apply algorithm.

I've implemented such policy because of issue description.
I guess I can replace that policy with `kAppendFittingSubpartitions` or add it as a choice.

Reply:
Got it.  Let's take what you have and file `kAppendFittingSubpartitions` (greedy) as a follow-up issue.  What you have is still very useful for the short term.  It be valuable to have both policies available, so I wouldn't replace what you have.

Reply:
Okay. Keeping it as it is.